TYPO3.Neos/Classes/TYPO3/Neos/Utility/NodeUriPathSegmentGenerator.php

```diff
@@ -31,22 +33,35 @@ class NodeUriPathSegmentGenerator
     protected $transliterationService;
 
     /**
+     * @Flow\Inject
+     * @var NodeTypeManager
+     */
+    protected $nodeTypeManager;
+
+    /**
+     * @Flow\Inject
+     * @var NodeSearchServiceInterface
+     */
+    protected $nodeSearchService;
+
+    /**
      * Sets the best possible uriPathSegment for the given Node.
      * Will use an already set uriPathSegment or alternatively the node name as base,
      * then checks if the uriPathSegment already exists on the same level and appends a counter until a unique path segment was found.
      *
      * @param NodeInterface $node
      * @return void
      */
-    public static function setUniqueUriPathSegment(NodeInterface $node)
+    public function setUniqueUriPathSegment(NodeInterface $node)
     {
         if ($node->getNodeType()->isOfType('TYPO3.Neos:Document')) {
-            $q = new FlowQuery(array($node));
-            $q = $q->context(array('invisibleContentShown' => true, 'removedContentShown' => true, 'inaccessibleContentShown' => true));
-
+            $parentNode = $node->getParent();
+            $documentNodeType = $this->nodeTypeManager->getNodeType('TYPO3.Neos:Document');
+            $context = $node->getContext();
             $possibleUriPathSegment = $initialUriPathSegment = !$node->hasProperty('uriPathSegment') ? $node->getName() : $node->getProperty('uriPathSegment');
+
             $i = 1;
-            while ($q->siblings('[instanceof TYPO3.Neos:Document][uriPathSegment="' . $possibleUriPathSegment . '"]')->count() > 0) {
+            while (count($this->nodeSearchService->findByProperties(['uriPathSegment' => $possibleUriPathSegment], [$documentNodeType], $context, $parentNode)) > 0) {
```

Comment:
so the `siblings` call doesn't take itself into account, but findByProperties does.. I think you need to check differently to make sure it doesn't just loop with itself.. just try it by updating a random properly on an existing node..